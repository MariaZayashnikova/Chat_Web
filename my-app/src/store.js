import { createStore, applyMiddleware } from 'redux'
import { persistStore, persistReducer } from 'redux-persist'
import storage from 'redux-persist/lib/storage'
import createSagaMiddleware from 'redux-saga'
import reducer from './reducer'
import rootSaga from './components/sagas'

const sagaMiddleware = createSagaMiddleware()

const persistConfig = {
    key: 'root',
    storage,
    whitelist: ['user'],
}
const persistedReducer = persistReducer(persistConfig, reducer)

let store = createStore(persistedReducer, applyMiddleware(sagaMiddleware))
let persistor = persistStore(store)
sagaMiddleware.run(rootSaga)

export { store, persistor }
