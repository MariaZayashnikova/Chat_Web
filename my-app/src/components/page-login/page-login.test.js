import React  from "react";
import { shallow, mount, render  } from 'enzyme';
import PageLogin from "./page-login";
import store from "../../store";

describe('Test Page Login', () => {
    it('Page Login have rendered correctly', () => {
        const pageLogin = render(<PageLogin store={store} />);
        expect(pageLogin).toMatchSnapshot();
    });
});