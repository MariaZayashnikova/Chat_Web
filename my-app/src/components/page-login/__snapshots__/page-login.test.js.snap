// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Page Login Page Login have rendered correctly 1`] = `
LoadedCheerio {
  "0": Node {
    "attribs": Object {
      "class": "PageLogin",
    },
    "children": Array [
      Node {
        "attribs": Object {
          "class": "TitleLogin",
        },
        "children": Array [
          Node {
            "data": "Chat",
            "next": null,
            "parent": [Circular],
            "prev": null,
            "type": "text",
          },
        ],
        "name": "h1",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": Node {
          "attribs": Object {
            "class": "containerForm",
          },
          "children": Array [
            Node {
              "attribs": Object {
                "class": "",
              },
              "children": Array [
                Node {
                  "attribs": Object {
                    "class": "input form-control",
                    "id": "email",
                    "name": "email",
                    "placeholder": "Email",
                    "type": "text",
                    "value": "",
                  },
                  "children": Array [],
                  "name": "input",
                  "namespace": "http://www.w3.org/1999/xhtml",
                  "next": Node {
                    "attribs": Object {
                      "class": "input form-control",
                      "id": "password",
                      "name": "password",
                      "placeholder": "Password",
                      "type": "password",
                      "value": "",
                    },
                    "children": Array [],
                    "name": "input",
                    "namespace": "http://www.w3.org/1999/xhtml",
                    "next": Node {
                      "attribs": Object {
                        "class": "btn btn-primary",
                        "type": "submit",
                      },
                      "children": Array [
                        Node {
                          "data": "Submit",
                          "next": null,
                          "parent": [Circular],
                          "prev": null,
                          "type": "text",
                        },
                      ],
                      "name": "button",
                      "namespace": "http://www.w3.org/1999/xhtml",
                      "next": null,
                      "parent": [Circular],
                      "prev": [Circular],
                      "type": "tag",
                      "x-attribsNamespace": Object {
                        "class": undefined,
                        "type": undefined,
                      },
                      "x-attribsPrefix": Object {
                        "class": undefined,
                        "type": undefined,
                      },
                    },
                    "parent": [Circular],
                    "prev": [Circular],
                    "type": "tag",
                    "x-attribsNamespace": Object {
                      "class": undefined,
                      "id": undefined,
                      "name": undefined,
                      "placeholder": undefined,
                      "type": undefined,
                      "value": undefined,
                    },
                    "x-attribsPrefix": Object {
                      "class": undefined,
                      "id": undefined,
                      "name": undefined,
                      "placeholder": undefined,
                      "type": undefined,
                      "value": undefined,
                    },
                  },
                  "parent": [Circular],
                  "prev": null,
                  "type": "tag",
                  "x-attribsNamespace": Object {
                    "class": undefined,
                    "id": undefined,
                    "name": undefined,
                    "placeholder": undefined,
                    "type": undefined,
                    "value": undefined,
                  },
                  "x-attribsPrefix": Object {
                    "class": undefined,
                    "id": undefined,
                    "name": undefined,
                    "placeholder": undefined,
                    "type": undefined,
                    "value": undefined,
                  },
                },
                Node {
                  "attribs": Object {
                    "class": "input form-control",
                    "id": "password",
                    "name": "password",
                    "placeholder": "Password",
                    "type": "password",
                    "value": "",
                  },
                  "children": Array [],
                  "name": "input",
                  "namespace": "http://www.w3.org/1999/xhtml",
                  "next": Node {
                    "attribs": Object {
                      "class": "btn btn-primary",
                      "type": "submit",
                    },
                    "children": Array [
                      Node {
                        "data": "Submit",
                        "next": null,
                        "parent": [Circular],
                        "prev": null,
                        "type": "text",
                      },
                    ],
                    "name": "button",
                    "namespace": "http://www.w3.org/1999/xhtml",
                    "next": null,
                    "parent": [Circular],
                    "prev": [Circular],
                    "type": "tag",
                    "x-attribsNamespace": Object {
                      "class": undefined,
                      "type": undefined,
                    },
                    "x-attribsPrefix": Object {
                      "class": undefined,
                      "type": undefined,
                    },
                  },
                  "parent": [Circular],
                  "prev": Node {
                    "attribs": Object {
                      "class": "input form-control",
                      "id": "email",
                      "name": "email",
                      "placeholder": "Email",
                      "type": "text",
                      "value": "",
                    },
                    "children": Array [],
                    "name": "input",
                    "namespace": "http://www.w3.org/1999/xhtml",
                    "next": [Circular],
                    "parent": [Circular],
                    "prev": null,
                    "type": "tag",
                    "x-attribsNamespace": Object {
                      "class": undefined,
                      "id": undefined,
                      "name": undefined,
                      "placeholder": undefined,
                      "type": undefined,
                      "value": undefined,
                    },
                    "x-attribsPrefix": Object {
                      "class": undefined,
                      "id": undefined,
                      "name": undefined,
                      "placeholder": undefined,
                      "type": undefined,
                      "value": undefined,
                    },
                  },
                  "type": "tag",
                  "x-attribsNamespace": Object {
                    "class": undefined,
                    "id": undefined,
                    "name": undefined,
                    "placeholder": undefined,
                    "type": undefined,
                    "value": undefined,
                  },
                  "x-attribsPrefix": Object {
                    "class": undefined,
                    "id": undefined,
                    "name": undefined,
                    "placeholder": undefined,
                    "type": undefined,
                    "value": undefined,
                  },
                },
                Node {
                  "attribs": Object {
                    "class": "btn btn-primary",
                    "type": "submit",
                  },
                  "children": Array [
                    Node {
                      "data": "Submit",
                      "next": null,
                      "parent": [Circular],
                      "prev": null,
                      "type": "text",
                    },
                  ],
                  "name": "button",
                  "namespace": "http://www.w3.org/1999/xhtml",
                  "next": null,
                  "parent": [Circular],
                  "prev": Node {
                    "attribs": Object {
                      "class": "input form-control",
                      "id": "password",
                      "name": "password",
                      "placeholder": "Password",
                      "type": "password",
                      "value": "",
                    },
                    "children": Array [],
                    "name": "input",
                    "namespace": "http://www.w3.org/1999/xhtml",
                    "next": [Circular],
                    "parent": [Circular],
                    "prev": Node {
                      "attribs": Object {
                        "class": "input form-control",
                        "id": "email",
                        "name": "email",
                        "placeholder": "Email",
                        "type": "text",
                        "value": "",
                      },
                      "children": Array [],
                      "name": "input",
                      "namespace": "http://www.w3.org/1999/xhtml",
                      "next": [Circular],
                      "parent": [Circular],
                      "prev": null,
                      "type": "tag",
                      "x-attribsNamespace": Object {
                        "class": undefined,
                        "id": undefined,
                        "name": undefined,
                        "placeholder": undefined,
                        "type": undefined,
                        "value": undefined,
                      },
                      "x-attribsPrefix": Object {
                        "class": undefined,
                        "id": undefined,
                        "name": undefined,
                        "placeholder": undefined,
                        "type": undefined,
                        "value": undefined,
                      },
                    },
                    "type": "tag",
                    "x-attribsNamespace": Object {
                      "class": undefined,
                      "id": undefined,
                      "name": undefined,
                      "placeholder": undefined,
                      "type": undefined,
                      "value": undefined,
                    },
                    "x-attribsPrefix": Object {
                      "class": undefined,
                      "id": undefined,
                      "name": undefined,
                      "placeholder": undefined,
                      "type": undefined,
                      "value": undefined,
                    },
                  },
                  "type": "tag",
                  "x-attribsNamespace": Object {
                    "class": undefined,
                    "type": undefined,
                  },
                  "x-attribsPrefix": Object {
                    "class": undefined,
                    "type": undefined,
                  },
                },
              ],
              "name": "form",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": null,
              "parent": [Circular],
              "prev": null,
              "type": "tag",
              "x-attribsNamespace": Object {
                "class": undefined,
              },
              "x-attribsPrefix": Object {
                "class": undefined,
              },
            },
          ],
          "name": "div",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": null,
          "parent": [Circular],
          "prev": [Circular],
          "type": "tag",
          "x-attribsNamespace": Object {
            "class": undefined,
          },
          "x-attribsPrefix": Object {
            "class": undefined,
          },
        },
        "parent": [Circular],
        "prev": null,
        "type": "tag",
        "x-attribsNamespace": Object {
          "class": undefined,
        },
        "x-attribsPrefix": Object {
          "class": undefined,
        },
      },
      Node {
        "attribs": Object {
          "class": "containerForm",
        },
        "children": Array [
          Node {
            "attribs": Object {
              "class": "",
            },
            "children": Array [
              Node {
                "attribs": Object {
                  "class": "input form-control",
                  "id": "email",
                  "name": "email",
                  "placeholder": "Email",
                  "type": "text",
                  "value": "",
                },
                "children": Array [],
                "name": "input",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": Node {
                  "attribs": Object {
                    "class": "input form-control",
                    "id": "password",
                    "name": "password",
                    "placeholder": "Password",
                    "type": "password",
                    "value": "",
                  },
                  "children": Array [],
                  "name": "input",
                  "namespace": "http://www.w3.org/1999/xhtml",
                  "next": Node {
                    "attribs": Object {
                      "class": "btn btn-primary",
                      "type": "submit",
                    },
                    "children": Array [
                      Node {
                        "data": "Submit",
                        "next": null,
                        "parent": [Circular],
                        "prev": null,
                        "type": "text",
                      },
                    ],
                    "name": "button",
                    "namespace": "http://www.w3.org/1999/xhtml",
                    "next": null,
                    "parent": [Circular],
                    "prev": [Circular],
                    "type": "tag",
                    "x-attribsNamespace": Object {
                      "class": undefined,
                      "type": undefined,
                    },
                    "x-attribsPrefix": Object {
                      "class": undefined,
                      "type": undefined,
                    },
                  },
                  "parent": [Circular],
                  "prev": [Circular],
                  "type": "tag",
                  "x-attribsNamespace": Object {
                    "class": undefined,
                    "id": undefined,
                    "name": undefined,
                    "placeholder": undefined,
                    "type": undefined,
                    "value": undefined,
                  },
                  "x-attribsPrefix": Object {
                    "class": undefined,
                    "id": undefined,
                    "name": undefined,
                    "placeholder": undefined,
                    "type": undefined,
                    "value": undefined,
                  },
                },
                "parent": [Circular],
                "prev": null,
                "type": "tag",
                "x-attribsNamespace": Object {
                  "class": undefined,
                  "id": undefined,
                  "name": undefined,
                  "placeholder": undefined,
                  "type": undefined,
                  "value": undefined,
                },
                "x-attribsPrefix": Object {
                  "class": undefined,
                  "id": undefined,
                  "name": undefined,
                  "placeholder": undefined,
                  "type": undefined,
                  "value": undefined,
                },
              },
              Node {
                "attribs": Object {
                  "class": "input form-control",
                  "id": "password",
                  "name": "password",
                  "placeholder": "Password",
                  "type": "password",
                  "value": "",
                },
                "children": Array [],
                "name": "input",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": Node {
                  "attribs": Object {
                    "class": "btn btn-primary",
                    "type": "submit",
                  },
                  "children": Array [
                    Node {
                      "data": "Submit",
                      "next": null,
                      "parent": [Circular],
                      "prev": null,
                      "type": "text",
                    },
                  ],
                  "name": "button",
                  "namespace": "http://www.w3.org/1999/xhtml",
                  "next": null,
                  "parent": [Circular],
                  "prev": [Circular],
                  "type": "tag",
                  "x-attribsNamespace": Object {
                    "class": undefined,
                    "type": undefined,
                  },
                  "x-attribsPrefix": Object {
                    "class": undefined,
                    "type": undefined,
                  },
                },
                "parent": [Circular],
                "prev": Node {
                  "attribs": Object {
                    "class": "input form-control",
                    "id": "email",
                    "name": "email",
                    "placeholder": "Email",
                    "type": "text",
                    "value": "",
                  },
                  "children": Array [],
                  "name": "input",
                  "namespace": "http://www.w3.org/1999/xhtml",
                  "next": [Circular],
                  "parent": [Circular],
                  "prev": null,
                  "type": "tag",
                  "x-attribsNamespace": Object {
                    "class": undefined,
                    "id": undefined,
                    "name": undefined,
                    "placeholder": undefined,
                    "type": undefined,
                    "value": undefined,
                  },
                  "x-attribsPrefix": Object {
                    "class": undefined,
                    "id": undefined,
                    "name": undefined,
                    "placeholder": undefined,
                    "type": undefined,
                    "value": undefined,
                  },
                },
                "type": "tag",
                "x-attribsNamespace": Object {
                  "class": undefined,
                  "id": undefined,
                  "name": undefined,
                  "placeholder": undefined,
                  "type": undefined,
                  "value": undefined,
                },
                "x-attribsPrefix": Object {
                  "class": undefined,
                  "id": undefined,
                  "name": undefined,
                  "placeholder": undefined,
                  "type": undefined,
                  "value": undefined,
                },
              },
              Node {
                "attribs": Object {
                  "class": "btn btn-primary",
                  "type": "submit",
                },
                "children": Array [
                  Node {
                    "data": "Submit",
                    "next": null,
                    "parent": [Circular],
                    "prev": null,
                    "type": "text",
                  },
                ],
                "name": "button",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": null,
                "parent": [Circular],
                "prev": Node {
                  "attribs": Object {
                    "class": "input form-control",
                    "id": "password",
                    "name": "password",
                    "placeholder": "Password",
                    "type": "password",
                    "value": "",
                  },
                  "children": Array [],
                  "name": "input",
                  "namespace": "http://www.w3.org/1999/xhtml",
                  "next": [Circular],
                  "parent": [Circular],
                  "prev": Node {
                    "attribs": Object {
                      "class": "input form-control",
                      "id": "email",
                      "name": "email",
                      "placeholder": "Email",
                      "type": "text",
                      "value": "",
                    },
                    "children": Array [],
                    "name": "input",
                    "namespace": "http://www.w3.org/1999/xhtml",
                    "next": [Circular],
                    "parent": [Circular],
                    "prev": null,
                    "type": "tag",
                    "x-attribsNamespace": Object {
                      "class": undefined,
                      "id": undefined,
                      "name": undefined,
                      "placeholder": undefined,
                      "type": undefined,
                      "value": undefined,
                    },
                    "x-attribsPrefix": Object {
                      "class": undefined,
                      "id": undefined,
                      "name": undefined,
                      "placeholder": undefined,
                      "type": undefined,
                      "value": undefined,
                    },
                  },
                  "type": "tag",
                  "x-attribsNamespace": Object {
                    "class": undefined,
                    "id": undefined,
                    "name": undefined,
                    "placeholder": undefined,
                    "type": undefined,
                    "value": undefined,
                  },
                  "x-attribsPrefix": Object {
                    "class": undefined,
                    "id": undefined,
                    "name": undefined,
                    "placeholder": undefined,
                    "type": undefined,
                    "value": undefined,
                  },
                },
                "type": "tag",
                "x-attribsNamespace": Object {
                  "class": undefined,
                  "type": undefined,
                },
                "x-attribsPrefix": Object {
                  "class": undefined,
                  "type": undefined,
                },
              },
            ],
            "name": "form",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": null,
            "parent": [Circular],
            "prev": null,
            "type": "tag",
            "x-attribsNamespace": Object {
              "class": undefined,
            },
            "x-attribsPrefix": Object {
              "class": undefined,
            },
          },
        ],
        "name": "div",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": null,
        "parent": [Circular],
        "prev": Node {
          "attribs": Object {
            "class": "TitleLogin",
          },
          "children": Array [
            Node {
              "data": "Chat",
              "next": null,
              "parent": [Circular],
              "prev": null,
              "type": "text",
            },
          ],
          "name": "h1",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": [Circular],
          "parent": [Circular],
          "prev": null,
          "type": "tag",
          "x-attribsNamespace": Object {
            "class": undefined,
          },
          "x-attribsPrefix": Object {
            "class": undefined,
          },
        },
        "type": "tag",
        "x-attribsNamespace": Object {
          "class": undefined,
        },
        "x-attribsPrefix": Object {
          "class": undefined,
        },
      },
    ],
    "name": "div",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": Node {
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
    },
    "prev": null,
    "type": "tag",
    "x-attribsNamespace": Object {
      "class": undefined,
    },
    "x-attribsPrefix": Object {
      "class": undefined,
    },
  },
  "_root": LoadedCheerio {
    "0": Node {
      "children": Array [
        Node {
          "attribs": Object {},
          "children": Array [
            Node {
              "attribs": Object {},
              "children": Array [],
              "name": "head",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": Node {
                "attribs": Object {},
                "children": Array [],
                "name": "body",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": null,
                "parent": [Circular],
                "prev": [Circular],
                "type": "tag",
                "x-attribsNamespace": Object {},
                "x-attribsPrefix": Object {},
              },
              "parent": [Circular],
              "prev": null,
              "type": "tag",
              "x-attribsNamespace": Object {},
              "x-attribsPrefix": Object {},
            },
            Node {
              "attribs": Object {},
              "children": Array [],
              "name": "body",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": null,
              "parent": [Circular],
              "prev": Node {
                "attribs": Object {},
                "children": Array [],
                "name": "head",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": [Circular],
                "parent": [Circular],
                "prev": null,
                "type": "tag",
                "x-attribsNamespace": Object {},
                "x-attribsPrefix": Object {},
              },
              "type": "tag",
              "x-attribsNamespace": Object {},
              "x-attribsPrefix": Object {},
            },
          ],
          "name": "html",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": null,
          "parent": [Circular],
          "prev": null,
          "type": "tag",
          "x-attribsNamespace": Object {},
          "x-attribsPrefix": Object {},
        },
      ],
      "name": "root",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-mode": "quirks",
    },
    "_root": [Circular],
    "length": 1,
    "options": Object {
      "decodeEntities": true,
      "xml": false,
    },
  },
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "xml": false,
  },
}
`;
